### PR TITLE
{chem,devel}[foss/2023a] f90wrap v0.2.13, MetalWalls v21.06.1

### PR DESCRIPTION
**Pull request by Crivella**
Depends on easyblock from https://github.com/easybuilders/easybuild-easyblocks/pull/3311

Adds the following EC files:

- MetalWalls (https://gitlab.com/ampere2/metalwalls)
- f90wrap (with a patch not yet included in the main distribution) as a dependency to build the python interface of MetalWalls

### CODE REVIEW

**Comment by ocaisa** [comment]
@boegelbot please test @ generoso
EB_ARGS="--include-easyblocks-from-pr 3311"

**Comment by ocaisa** [comment]
@boegelbot please test @ jsc-zen3
EB_ARGS="--include-easyblocks-from-pr 3311"

**Comment by ocaisa** [comment]
@boegelbot please test @ generoso
EB_ARGS="--include-easyblocks-from-pr 3311"

**Comment by ocaisa** [comment]
@boegelbot please test @ jsc-zen3
EB_ARGS="--include-easyblocks-from-pr 3311"

**Review by ocaisa — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/f/f90wrap/f90wrap-0.2.13-foss-2023a.eb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+easyblock = 'PythonPackage'
+
+name = 'f90wrap'
+version = '0.2.13'
```
[the remark this answers is not in the document]
Comment (Crivella) [reply easybuild/easyconfigs/f/f90wrap/f90wrap-0.2.13-foss-2023a.eb:4]: Just tested, 0.2.14 does compile without the patch, but MetalWalls fails to compile with many errors such as reported below.

Should Iadd a dependency check for f90wraps <= 0.2.13 in the EB for MetalWalls?

```
   87 | subroutine f90wrap_mw_configuration_line__mw_configuration_line_t_initialise(this)
      |                                                                            1
Error: Name at (1) is too long
f90wrap_configuration_line.f90:98:3:

   98 | end subroutine f90wrap_mw_configuration_line__mw_configuration_line_t_initialise
      |   1
Error: Expecting END PROGRAM statement at (1)
f90wrap_configuration_line.f90:101:60:

  101 |     use mw_configuration_line, only: mw_configuration_line_t
      |                                                            1
Error: Unexpected USE statement at (1)
f90wrap_configuration_line.f90:102:17:

  102 |     implicit none
      |                 1
Error: Duplicate IMPLICIT NONE statement at (1)
f90wrap_configuration_line.f90:104:41:

  104 |     type mw_configuration_line_t_ptr_type
      |                                         1
Error: Derived type definition of mw_configuration_line_t_ptr_type at (1) has already been defined
f90wrap_configuration_line.f90:105:61:

  105 |         type(mw_configuration_line_t), pointer :: p => NULL()
      |                                                             1
Error: Unexpected data declaration statement at (1)
f90wrap_configuration_line.f90:106:7:

  106 |     end type mw_configuration_line_t_ptr_type
      |       1
Error: Expecting END PROGRAM statement at (1)
f90wrap_configuration_line.f90:107:54:
...
```

[the remark this answers is not in the document]
Comment (ocaisa) [reply easybuild/easyconfigs/f/f90wrap/f90wrap-0.2.13-foss-2023a.eb:4]: Ok, let's kick the can down the road on this. I don't know when Metalwalls will be getting an update but they will have to deal with this

**Review by ocaisa — APPROVED** [review]

File: easybuild/easyconfigs/f/f90wrap/f90wrap-0.2.13_fix_old_typing.patch [hunk 2]
```diff
@@ -0,0 +1,36 @@
+--- f90wrap/arraydatamodule.c.orig	2024-04-19 14:22:57.488265190 +0200
```
Comment (ocaisa) [inline easybuild/easyconfigs/f/f90wrap/f90wrap-0.2.13_fix_old_typing.patch:1]: Patch should have a description above this line

**Review by ocaisa — CHANGES_REQUESTED** [review]
CI is failing

**Review by ocaisa — DISMISSED** [review]

**Comment by ocaisa** [comment]
@boegelbot please test @ jsc-zen3
EB_ARGS="--include-easyblocks-from-pr 3311"

**Comment by ocaisa** [comment]
@boegelbot please test @ generoso
EB_ARGS="--include-easyblocks-from-pr 3311"

**Pull request merged**